python/ql/src/experimental/Security/CWE-285/PamAuthorization.ql

```diff
@@ -0,0 +1,58 @@
+/**
+ * @name Authorization bypass due to incorrect usage of PAM
+ * @description Using only the `pam_authenticate` call to check the validity of a login can lead to a authorization bypass.
+ * @kind problem
+ * @problem.severity warning
+ * @id py/pam-auth-bypass
+ * @tags security
+ *       external/cwe/cwe-285
+ */
+
+import python
+import semmle.python.ApiGraphs
+import experimental.semmle.python.Concepts
+import semmle.python.dataflow.new.TaintTracking
+
+private class LibPam extends API::Node {
+  LibPam() {
```

Comment:
Instead of making this a class, please make it a predicate, such as

```codeql
API::Node libPam() {
  exists( ... | ... | result = cdll_call.getReturn() )
}
```